shared/js/passcode_helper.js

```diff
@@ -10,154 +10,128 @@
   const DEFAULT_ALGORITHM = 'SHA-1'; //XXX Update to SHA-256 after bug 554827?
   const DEFAULT_ITERATIONS = 1000;
 
-  var PasscodeHelper = function() {
-    /* Usage:
-     *      var ph = new PasscodeHelper();
-     *      ph.setPassccode(string) -> Promise => digest (or false)
-     *      ph.checkPasscode(string) -> Promise resolves to a boolean
-     * */
-
-    this._iterations = DEFAULT_ITERATIONS;
-    this._algorithm = DEFAULT_ALGORITHM;
-    this._testPass = null;
-    this._salt = null;
-  };
-
-
-  PasscodeHelper.prototype._encode = function(str) {
-      return new TextEncoder('utf-8').encode(str);
-    };
-  PasscodeHelper.prototype._toTypedArray = function(obj) {
-      // SettingsAPI doesnt like arrays and gives us an Object { 1: .., 2: .. }
-      var a = [];
-      for (var key in obj) {
-        a.push(obj[key]);
-      }
-      return new Uint8Array(a);
-    };
-  PasscodeHelper.prototype._deriveBits = function(pwKey, salt,
-                                                  iterations, algorithm) {
-      var length = 256;
-      var params = {
-        name: 'PBKDF2',
-        hash: algorithm,
-        salt: salt,
-        iterations: iterations
-      };
-      return crypto.subtle.deriveBits(params, pwKey, length);
+  function _encode(str) {
+    return new TextEncoder('utf-8').encode(str);
+  }
+  function _toTypedArray(obj) {
+    // SettingsAPI doesnt like arrays and gives us an Object { 1: .., 2: .. }
+    var a = [];
+    for (var key in obj) {
+      a.push(obj[key]);
+    }
+    return new Uint8Array(a);
+  }
+  function _deriveBits(pwKey, salt, iterations, algorithm) {
+    var length = 256;
```

Comment:
This is twofold, right?
First, move this up to the other constants (line 6-11) and secondly, changing this to 160, since SHA-1 doesn't even gives us that much.